src/benchq/resource_estimation/graph/graph_estimator.py

```diff
@@ -84,13 +85,16 @@ def __init__(
         decoder_model: Optional[DecoderModel] = None,
         optimization: str = "space",
         substrate_scheduler_preset: str = "fast",
+        widget_iterable_factory: Callable[
+            [BasicArchitectureModel], Iterable[Widget]
+        ] = default_widget_list,
```

Comment:
Is there any easier way to do this?? Factories for iterables only make the code much less readable...

Reply:
I can imagine this being complicated. Ive changed it to just Iterable[Widget] for anyone who wants to provide their own widget list, with default being chosen from default_widget_list based on hw_model.